server/src/main/java/com/saralein/server/middleware/StaticMiddleware.java

```diff
@@ -0,0 +1,81 @@
+package com.saralein.server.middleware;
+
+import com.saralein.server.FileHelper;
+import com.saralein.server.controller.Controller;
+import com.saralein.server.controller.DirectoryController;
+import com.saralein.server.controller.FileController;
+import com.saralein.server.protocol.Methods;
+import com.saralein.server.request.Request;
+import com.saralein.server.response.Response;
+import com.saralein.server.router.Router;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class StaticMiddleware {
+    private final FileHelper fileHelper;
+    private final Router router;
+    private final Controller directoryController;
+    private final Controller fileController;
+
+    public StaticMiddleware(Path root, Router router) {
+        this(new FileHelper(root), router, new DirectoryController(new FileHelper(root)),
+                new FileController(new FileHelper(root)));
+    }
```

Comment:
Seems like all the file related stuff falls in here, it feels like partial content belongs in here too.

Reply:
Yeah, it will go here.  It doesn't exist on this branch yet though.  Perhaps I should have waited for it to open this.

Reply:
Yeah good point, I guess once one gets merged the second'll get rebased.